docs/reference/schemas/config/resource.md

```diff
@@ -80,30 +84,57 @@ Required: true
 To declare that a resource instance is dependent on another instance in the configuration, define
 the `dependsOn` property.
 
-This property must be an array of dependency declarations. Each dependency must use this
-syntax:
+This property defines a list of DSC Resource instances that DSC must successfully process before
+processing this instance. Each value for this property must be the `resourceID()` lookup for
+another instance in the configuration. Multiple instances can depend on the same instance, but
+every dependency for an instance must be unique in that instance's `dependsOn` property.
+
+The `resourceID()` function uses this syntax:
 
 ```yaml
-"[<resource-type-name>]<instance-name>"
+"[resourceId('<resource-type-name>', '']<instance-name>"
 ```
 
-In the dependency syntax, `<resource-type-name>` is the `type` property of the dependent resource
-and `<instance-name>` is the dependency's `name` property.
+The `<resource-type-name>` value is the `type` property of the dependent resource and
+`<instance-name>` is the dependency's `name` property. When adding a dependency in a YAML-format
+configuration document, always wrap the `resourceID()` lookup in double quotes (`"`).
+
+For example, this instance depends on an instance of the `Microsoft.Windows/Registry`
+resource named `Tailspin Key`:
+
+```yaml
+- name: Update Tailspin Automatically
+  type: Microsoft.Windows/Registry
+  properties:
+    keyPath:   HKCU\tailspin\updates
+    valueName: automatic
+    valueData:
+      String: enable
+  dependsOn:
+    - "[resourceId('Microsoft.Windows/Registry', 'Tailspin Key')]"
+```
 
-Multiple instances can depend on the same instance, but every dependency for an instance must be
-unique in that instance's `dependsOn` property.
+> [!NOTE]
+> When defining dependencies for [nested resource instances][02], instances can only reference
+> dependencies in the same resource provider or group instance. They can't use the `resourceId()`
+> function to lookup instances at the top-level of the configuration document or inside another
+> provider or group instance.
+>
+> If a top-level instance depends on a nested instance, use the `resourceId()` function to lookup
+> the instance of the provider or group containing the dependency instance instead.
```

Comment:
Does it make sense to say that if there is a need to depend on something deeply nested, you can use grouping to achieve the same effect.  An example would probably be best.

Reply:
Can you share a pseudo-example of what you mean? I can convert this into a full example afterward, though that may better belong in the eventual conceptual article for managing dependencies in configuration documents.